internal/storage/bucket_handle_test.go

```diff
@@ -0,0 +1,105 @@
+// Copyright 2022 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package storage
+
+import (
+	"context"
+	"testing"
+
+	"github.com/fsouza/fake-gcs-server/fakestorage"
+	"github.com/jacobsa/gcloud/gcs"
+	. "github.com/jacobsa/ogletest"
+)
+
+const fileContent string = "hello gcsfuse"
+const validFilePathInBucket string = "some/object/file.txt"
+const invalidFilePathInBucket string = "test/foo"
+
+func TestBucketHandle(t *testing.T) { RunTests(t) }
+
+type BucketHandleTest struct {
+	fakeStorageServer *fakestorage.Server
+}
+
+var _ SetUpInterface = &BucketHandleTest{}
+var _ TearDownInterface = &BucketHandleTest{}
+
+func init() { RegisterTestSuite(&BucketHandleTest{}) }
+
+func (t *BucketHandleTest) SetUp(_ *TestInfo) {
+	var err error
+	t.fakeStorageServer, err = fakestorage.NewServerWithOptions(fakestorage.Options{
```

Comment:
please check if you can create a seperate file to handle all fakeserver creation logic and methods to insert/delete data which can be used by other test.

Reply:
I have created the file and add the basic methods (GetDefaultObject and CreateServer).